frontends/p4/typeChecking/typeChecker.cpp

```diff
@@ -1144,6 +1153,11 @@ const IR::Node* TypeInference::postorder(IR::Type_Set* type) {
     return type;
 }
 
+const IR::Node* TypeInference::postorder(IR::Type_ValueSet* type) {
+    (void)setTypeType(type);
```

Comment:
Can you try to set this to `Type_Set(elementType)` and see if it works?
The you won't need to do anything in the select expression.
This will be a funny type declaration, because the type of the declaration is not a TypeType of itself.
Also, the case in canonicalize won't be necessary.

Reply:
Like this? 
```
const IR::Node* TypeInference::postorder(IR::Type_ValueSet* type) {
    auto tt = new IR::Type_Set(type->elementType);
    (void)setTypeType(tt);
    return tt;
}
```

Reply:
This may work.